pkg/core/blockchain.go

```diff
@@ -782,6 +782,15 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 		// because changes applied are the ones from HALTed transactions.
 		return fmt.Errorf("error while trying to apply MPT changes: %w", err)
 	}
+	if bc.GetConfig().StateRootInHeader && bc.HeaderHeight() > sr.Index {
```

Comment:
Please use `bc.config` directly here, we don't need a copy of it (it's for external users).